src/cargo/ops/cargo_rustc/context.rs

```diff
@@ -339,22 +339,22 @@ impl<'a, 'b: 'a> Context<'a, 'b> {
         };
         let mut ret = deps.map(|id| self.get_package(id)).filter(|dep| {
             let pkg_dep = pkg.dependencies().iter().find(|d| {
-                d.name() == dep.name()
-            }).unwrap();
-
-            // If this target is a build command, then we only want build
-            // dependencies, otherwise we want everything *other than* build
-            // dependencies.
-            let is_correct_dep = target.is_custom_build() == pkg_dep.is_build();
-
-            // If this dependency is *not* a transitive dependency, then it
-            // only applies to test/example targets
-            let is_actual_dep = pkg_dep.is_transitive() ||
```

Comment:
Could you keep these variable names for now? Reading large multi-line conditionals is sometimes a little tough.